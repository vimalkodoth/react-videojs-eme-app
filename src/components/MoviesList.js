import React, { Component } from "react";
import { connect } from "react-redux";
import { fetchMoviesList } from "./../actionCreators";
import List from "./List";
import PortraitItem from "./PortraitItem";
import { css } from "@emotion/core";
import { withRouter } from "react-router";
import { array, object, func } from "prop-types";
class MoviesList extends Component {
    static propTypes = {
        moviesList: array,
        fetchMoviesList: func,
        location: object
    };
    static defaultProps = {
        moviesList: [],
        location: {}
    };
    componentDidMount() {
        if (this.props.moviesList.length) return;
        this.props.fetchMoviesList();
    }

    render() {
        const { location } = this.props;
        if (!this.props.moviesList) return;
        return this.props.moviesList.map((list) => {
            return (
                //Flexible Compound Component
                <div css={MoviesListStyles} key={list.id}>
                    <List key={list.id}>
                        <List.Title>
                            <h2>{list.name}</h2>
                        </List.Title>
                        <List.Horizontal>
                            {list.contents.data.map((movieItem) => {
                                return (
                                    <PortraitItem
                                        key={movieItem.id}
                                        item={movieItem}
                                        to={{
                                            pathname: `/details/${movieItem.id}`,
                                            state: { from: location.pathname }
                                        }}
                                    ></PortraitItem>
                                );
                            })}
                        </List.Horizontal>
                    </List>
                </div>
            );
        });
    }
}

const mapStateToProps = (state) => ({ moviesList: state.moviesList });
const mapDispatchToProps = (dispatch) => ({
    fetchMoviesList() {
        dispatch(fetchMoviesList());
    }
});

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(withRouter(MoviesList));

const MoviesListStyles = css`
    & {
        padding: 0 5px 15px;
        @media (min-width: 560px) {
            padding: 0 15px 25px;
        }
    }
    & h2 {
        font-size: 1em;
        @media (min-width: 880px) {
            font-size: 1.5em;
        }
        margin-left: 5px;
    }
`;
