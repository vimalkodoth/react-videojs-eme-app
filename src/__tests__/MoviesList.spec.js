import React from "react";
import { render, mount } from "enzyme";
import { MemoryRouter } from "react-router-dom";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import MoviesList from "./../components/MoviesList";
import list from "./json/list.json";
const mockStore = configureStore([]);
describe("MoviesList", () => {
    let store;
    const originalConsoleError = console.error;
    beforeEach(() => {
        console.error = jest.fn();
    });

    afterEach(() => {
        console.error = originalConsoleError;
        store = mockStore({});
    });

    it("renders correctly", () => {
        const childCount = 3;
        store = mockStore({
            moviesList: [list.data]
        });
        const component = render(
            <Provider store={store}>
                <MemoryRouter>
                    <MoviesList />
                </MemoryRouter>
            </Provider>
        );
        expect(component).toMatchSnapshot();
        expect(component.find('[data-testid="h-list"]').length).toEqual(1);
        expect(
            component.find('[data-testid="h-list"]').children().length
        ).toEqual(childCount);
    });

    it("calls dispatch on mount", () => {
        store.dispatch = jest.fn();
        const component = mount(
            <Provider store={store}>
                <MemoryRouter>
                    <MoviesList />
                </MemoryRouter>
            </Provider>
        );
        expect(store.dispatch).toHaveBeenCalledTimes(1);
    });
});
