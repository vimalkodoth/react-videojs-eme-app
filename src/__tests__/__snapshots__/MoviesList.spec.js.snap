// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesList renders correctly 1`] = `
<div
  class="css-k0kzju-MoviesListStyles"
>
  <div
    class="css-1htuyjk-ListStyle"
  >
    <h2>
      CINEMA - Ãšltimos estrenos
    </h2>
  </div>
  <div
    class="css-1htuyjk-ListStyle"
  >
    <div
      class="css-1ae81ju-List"
      data-testid="h-list"
    >
      <ul
        class="css-1qs8dlz-ListInner"
        style="transform:translateX(0%)"
      >
        <li
          class="css-1puajek-ListItem"
          data-item="0"
        >
          <div
            class="item css-qbkfep-PortraitItemStyles"
          >
            <a
              href="/details/mulan-718ddc0b-8c5f-4afd-8071-13592e4e21a1"
            >
              <img
                class="css-ucw8vg-Image"
                src="https://images-0.wuaki.tv/system/artworks/134332/master/mulan-2020-1601313907.jpeg"
              />
              <img
                class="css-4pvplo-Image"
                src="[object Object]"
              />
            </a>
          </div>
        </li>
        <li
          class="css-1puajek-ListItem"
          data-item="1"
        >
          <div
            class="item css-qbkfep-PortraitItemStyles"
          >
            <a
              href="/details/the-sleeping-beauty-ballet"
            >
              <img
                class="css-ucw8vg-Image"
                src="https://images-1.wuaki.tv/system/artworks/135881/master/the-sleeping-beauty-ballet-1602253454.jpeg"
              />
              <img
                class="css-4pvplo-Image"
                src="[object Object]"
              />
            </a>
          </div>
        </li>
        <li
          class="css-1puajek-ListItem"
          data-item="2"
        >
          <div
            class="item css-qbkfep-PortraitItemStyles"
          >
            <a
              href="/details/la-fuerza-de-la-naturaleza"
            >
              <img
                class="css-ucw8vg-Image"
                src="https://images-1.wuaki.tv/system/artworks/130826/master/la-fuerza-de-la-naturaleza-1600249838.jpeg"
              />
              <img
                class="css-4pvplo-Image"
                src="[object Object]"
              />
            </a>
          </div>
        </li>
        <li
          class="css-1puajek-ListItem"
          data-item="3"
        >
          <div
            class="item css-qbkfep-PortraitItemStyles"
          >
            <a
              href="/details/atrapado"
            >
              <img
                class="css-ucw8vg-Image"
                src="https://images-0.wuaki.tv/system/artworks/134275/master/atrapado-1601552390.jpeg"
              />
              <img
                class="css-4pvplo-Image"
                src="[object Object]"
              />
            </a>
          </div>
        </li>
        <li
          class="css-1puajek-ListItem"
          data-item="4"
        >
          <div
            class="item css-qbkfep-PortraitItemStyles"
          >
            <a
              href="/details/a-vigilante"
            >
              <img
                class="css-ucw8vg-Image"
                src="https://images-0.wuaki.tv/system/artworks/132700/master/a-vigilante-1599127920.jpeg"
              />
              <img
                class="css-4pvplo-Image"
                src="[object Object]"
              />
            </a>
          </div>
        </li>
        <li
          class="css-1puajek-ListItem"
          data-item="5"
        >
          <div
            class="item css-qbkfep-PortraitItemStyles"
          >
            <a
              href="/details/deep-blue-sea-3"
            >
              <img
                class="css-ucw8vg-Image"
                src="https://images-2.wuaki.tv/system/artworks/127502/master/deep-blue-sea-3-1598947260.jpeg"
              />
              <img
                class="css-4pvplo-Image"
                src="[object Object]"
              />
            </a>
          </div>
        </li>
        <li
          class="css-1puajek-ListItem"
          data-item="6"
        >
          <div
            class="item css-qbkfep-PortraitItemStyles"
          >
            <a
              href="/details/la-influencia"
            >
              <img
                class="css-ucw8vg-Image"
                src="https://images-1.wuaki.tv/system/artworks/127623/master/la-influencia-1601890708.jpeg"
              />
              <img
                class="css-4pvplo-Image"
                src="[object Object]"
              />
            </a>
          </div>
        </li>
        <li
          class="css-1puajek-ListItem"
          data-item="7"
        >
          <div
            class="item css-qbkfep-PortraitItemStyles"
          >
            <a
              href="/details/la-herencia"
            >
              <img
                class="css-ucw8vg-Image"
                src="https://images-2.wuaki.tv/system/artworks/130827/master/la-herencia-1595322451.jpeg"
              />
              <img
                class="css-4pvplo-Image"
                src="[object Object]"
              />
            </a>
          </div>
        </li>
        <li
          class="css-1puajek-ListItem"
          data-item="8"
        >
          <div
            class="item css-qbkfep-PortraitItemStyles"
          >
            <a
              href="/details/escape-plan-3"
            >
              <img
                class="css-ucw8vg-Image"
                src="https://images-0.wuaki.tv/system/artworks/132698/master/escape-plan-3-1597237664.jpeg"
              />
              <img
                class="css-4pvplo-Image"
                src="[object Object]"
              />
            </a>
          </div>
        </li>
        <li
          class="css-1puajek-ListItem"
          data-item="9"
        >
          <div
            class="item css-qbkfep-PortraitItemStyles"
          >
            <a
              href="/details/mortal"
            >
              <img
                class="css-ucw8vg-Image"
                src="https://images-1.wuaki.tv/system/artworks/132704/master/mortal-1597848971.jpeg"
              />
              <img
                class="css-4pvplo-Image"
                src="[object Object]"
              />
            </a>
          </div>
        </li>
        <li
          class="css-1puajek-ListItem"
          data-item="10"
        >
          <div
            class="item css-qbkfep-PortraitItemStyles"
          >
            <a
              href="/details/el-avion-del-dinero"
            >
              <img
                class="css-ucw8vg-Image"
                src="https://images-1.wuaki.tv/system/artworks/131442/master/money-plane-1596784853.jpeg"
              />
              <img
                class="css-4pvplo-Image"
                src="[object Object]"
              />
            </a>
          </div>
        </li>
        <li
          class="css-1puajek-ListItem"
          data-item="11"
        >
          <div
            class="item css-qbkfep-PortraitItemStyles"
          >
            <a
              href="/details/2-22-la-hora-senalada"
            >
              <img
                class="css-ucw8vg-Image"
                src="https://images-3.wuaki.tv/system/artworks/132701/master/2-22-la-hora-senalada-1597241725.jpeg"
              />
              <img
                class="css-4pvplo-Image"
                src="[object Object]"
              />
            </a>
          </div>
        </li>
        <li
          class="css-1puajek-ListItem"
          data-item="12"
        >
          <div
            class="item css-qbkfep-PortraitItemStyles"
          >
            <a
              href="/details/la-lista-de-a-la-mierda"
            >
              <img
                class="css-ucw8vg-Image"
                src="https://images-2.wuaki.tv/system/artworks/135548/master/la-lista-de-a-la-mierda-1601555234.jpeg"
              />
              <img
                class="css-4pvplo-Image"
                src="[object Object]"
              />
            </a>
          </div>
        </li>
        <li
          class="css-1puajek-ListItem"
          data-item="13"
        >
          <div
            class="item css-qbkfep-PortraitItemStyles"
          >
            <a
              href="/details/el-caso-collini"
            >
              <img
                class="css-ucw8vg-Image"
                src="https://images-3.wuaki.tv/system/artworks/132703/master/el-caso-collini-1599116175.jpeg"
              />
              <img
                class="css-4pvplo-Image"
                src="[object Object]"
              />
            </a>
          </div>
        </li>
        <li
          class="css-1puajek-ListItem"
          data-item="14"
        >
          <div
            class="item css-qbkfep-PortraitItemStyles"
          >
            <a
              href="/details/el-tunel"
            >
              <img
                class="css-ucw8vg-Image"
                src="https://images-2.wuaki.tv/system/artworks/130825/master/el-tunel-1597846402.png"
              />
              <img
                class="css-4pvplo-Image"
                src="[object Object]"
              />
            </a>
          </div>
        </li>
        <li
          class="css-1puajek-ListItem"
          data-item="15"
        >
          <div
            class="item css-qbkfep-PortraitItemStyles"
          >
            <a
              href="/details/oro-blanco"
            >
              <img
                class="css-ucw8vg-Image"
                src="https://images-2.wuaki.tv/system/artworks/130529/master/oro-blanco-1596785227.jpeg"
              />
              <img
                class="css-4pvplo-Image"
                src="[object Object]"
              />
            </a>
          </div>
        </li>
        <li
          class="css-1puajek-ListItem"
          data-item="16"
        >
          <div
            class="item css-qbkfep-PortraitItemStyles"
          >
            <a
              href="/details/desconectados"
            >
              <img
                class="css-ucw8vg-Image"
                src="https://images-3.wuaki.tv/system/artworks/133873/master/desconectados-1600768252.jpeg"
              />
              <img
                class="css-4pvplo-Image"
                src="[object Object]"
              />
            </a>
          </div>
        </li>
        <li
          class="css-1puajek-ListItem"
          data-item="17"
        >
          <div
            class="item css-qbkfep-PortraitItemStyles"
          >
            <a
              href="/details/corre-como-una-chica"
            >
              <img
                class="css-ucw8vg-Image"
                src="https://images-1.wuaki.tv/system/artworks/132705/master/corre-como-una-chica-1598530917.jpeg"
              />
              <img
                class="css-4pvplo-Image"
                src="[object Object]"
              />
            </a>
          </div>
        </li>
      </ul>
      <div
        class="css-7x77wu-ButtonStyles"
        data-testid="button"
        style="display:none"
      />
      <div
        class="css-1gr9e02-ButtonStyles"
        data-testid="button"
        style="display:block"
      />
    </div>
  </div>
</div>
`;
